### PR TITLE
thunderscope:  simple pass generator layer and simulated test fix

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
* added a layer to visualize the top 3 passes from the pass generator
* create conversion functions for them
* fixed simulated tests to LOG(VISUALIZE) world so we see it in thunderscope

### Testing Done
manually tested

you can try it for your self by running

`bazel run //software/thunderscope:thunderscope`

followed by

`bazel run //software/ai/hl/stp/play/shoot_or_pass:shoot_or_pass_play_test --cache_test_results=no --test_arg="--run_sim_in_realtime"`

### Resolved Issues
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/simulated_tests/simulated_test_fixture.cpp [hunk 1]
```diff
@@ -98,6 +99,10 @@ void SimulatedTestFixture::SetUp()
     {
         enableVisualizer();
     }
+    if (TbotsGtestMain::run_sim_in_realtime)
+    {
+        run_simulation_in_realtime = true;
+    }
```
Comment (jonathanlew) [inline src/software/simulated_tests/simulated_test_fixture.cpp:105]: what does this do exactly?

File: src/software/thunderscope/field/pass_layer.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import software.thunderscope.constants as constants
+from pyqtgraph.Qt import QtCore, QtGui
+from proto.visualization_pb2 import PassVisualization
+from software.networking.threaded_unix_listener import ThreadedUnixListener
+
+
+class PassLayer(FieldLayer):
+
+    PASS_VISUALIZATION_TIMEOUT_S = 0.5
+    NUM_TOP_PASSES_TO_RENDER = 3
```
Comment (jonathanlew) [inline src/software/thunderscope/field/pass_layer.py:13]: shouldn't we have some global timeout for AI drawing things?

**Review by jonathanlew — COMMENTED** [review]
Cool you got this working. It's hard to tell which circles are the obstacle and which ones are the passes

**Comment by akhilveeraghanta** [comment]
going to close this for now, don't have enough bandwidth and i think we can find a better way to plot the passes (because its hard to tell with just circles)

**Pull request closed without merging**